src/components/views/elements/ImageView.tsx

```diff
@@ -160,41 +160,82 @@ export default class ImageView extends React.Component<IProps, IState> {
         });
     };
 
-    private zoom(delta: number) {
-        const newZoom = this.state.zoom + delta;
+    private zoomDelta(delta: number, zoomX?: number, zoomY?: number) {
+        this.zoom(this.state.zoom + delta, zoomX, zoomY);
+    }
+
+    private zoom(zoomLevel: number, zoomX?: number, zoomY?: number) {
+        const oldZoom = this.state.zoom;
+        const newZoom = Math.min(zoomLevel, this.state.maxZoom);
 
         if (newZoom <= this.state.minZoom) {
+            // Zoom out fully
             this.setState({
                 zoom: this.state.minZoom,
                 translationX: 0,
                 translationY: 0,
             });
-            return;
-        }
-        if (newZoom >= this.state.maxZoom) {
-            this.setState({ zoom: this.state.maxZoom });
-            return;
-        }
+        } else if (zoomX === undefined && zoomY === undefined) {
```

Comment:
Because they could be `0` if you're trying to zoom in on the top-left corner of the image

Reply:
:man_facepalming:

Reply:
In that case shall we go with 

```suggestion
        } else if (typeof zoomX !== "number" && typeof zoomY !== "number") {
```

I believe that TypeScript would allow you to pass `null` as a value down here